components/server/src/user/user-service.ts

```diff
@@ -421,10 +420,27 @@ export class UserService {
         // update user
         user.name = user.name || authUser.name || authUser.primaryEmail;
         user.avatarUrl = user.avatarUrl || authUser.avatarUrl;
-
+        await this.onAfterUserLoad(user);
         await this.updateUserIdentity(user, candidate, token);
     }
 
+    async onAfterUserLoad(user: User): Promise<User> {
+        try {
+            let result = user;
```

Comment:
This is a bit confusing, because I think we're using the same object during to apply the migration to, anyway.